You should not include Windows specific headers when building under Cygwin

In [xxhsum.c][1], you define some Windows specific constructs when building under Cygwin. That is not correct. The purpose of Cygwin is to provide a POSIX API layer over Windows APIs (whereas the purpose of MinGW* projects is to allow Windows programs to be compiled using the GCC toolchain).

```text
$ gcc -DXXH_PRIVATE_API -O3 -march=native xxhsum.c -o xxhsum-cygwin.exe
xxhsum.c: In function ‘BMK_hash’:
xxhsum.c:71:42: warning: implicit declaration of function ‘_fileno’ [-Wimplicit-function-declaration]
 #  define SET_BINARY_MODE(file) _setmode(_fileno(file), _O_BINARY)
                                          ^
xxhsum.c:523:9: note: in expansion of macro ‘SET_BINARY_MODE’
         SET_BINARY_MODE(stdin);
         ^
xxhsum.c:71:57: error: ‘_O_BINARY’ undeclared (first use in this function)
 #  define SET_BINARY_MODE(file) _setmode(_fileno(file), _O_BINARY)
                                                         ^
xxhsum.c:523:9: note: in expansion of macro ‘SET_BINARY_MODE’
         SET_BINARY_MODE(stdin);
         ^
xxhsum.c:71:57: note: each undeclared identifier is reported only once for each function it appears in
 #  define SET_BINARY_MODE(file) _setmode(_fileno(file), _O_BINARY)
                                                         ^
xxhsum.c:523:9: note: in expansion of macro ‘SET_BINARY_MODE’
         SET_BINARY_MODE(stdin);
         ^
xxhsum.c: In function ‘main’:
xxhsum.c:72:33: warning: implicit declaration of function ‘_isatty’ [-Wimplicit-function-declaration]
 #  define IS_CONSOLE(stdStream) _isatty(_fileno(stdStream))
                                 ^
xxhsum.c:1226:33: note: in expansion of macro ‘IS_CONSOLE’
     if ( (filenamesStart==0) && IS_CONSOLE(stdin) ) return badusage(exename);
```

With the following change:

```diff
$ git diff
diff --git a/xxhsum.c b/xxhsum.c
index 1928141..322cfd1 100644
--- a/xxhsum.c
+++ b/xxhsum.c
@@ -62,7 +62,7 @@
 /*-************************************
 *  OS-Specific Includes
 **************************************/
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__)
+#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(_WIN32)
 #  include <fcntl.h>    /* _O_BINARY */
 #  include <io.h>       /* _setmode, _isatty */
 #  ifdef __MINGW32__
```

We get:

```text
$ gcc -DXXH_PRIVATE_API -O3 -march=native xxhsum.c -o xxhsum-cygwin
$ ./xxhsum-cygwin -h
./xxhsum-cygwin 0.6.2 (64-bits little endian), by Yann Collet
...
```

HTH.

[1]: https://github.com/Cyan4973/xxHash/blob/7fc5ce8592f10c68506f4a070bfa6a9ab2004b44/xxhsum.c#L65

--- a/xxhsum.c
+++ b/xxhsum.c
@@ -93,7 +93,7 @@
  || defined(__MSYS__)
 #  include <unistd.h>   /* isatty */
 #  define IS_CONSOLE(stdStream) isatty(fileno(stdStream))
-#elif defined(MSDOS) || defined(OS2) || defined(__CYGWIN__)
+#elif defined(MSDOS) || defined(OS2)
 #  include <io.h>       /* _isatty */
 #  define IS_CONSOLE(stdStream) _isatty(_fileno(stdStream))
 #elif defined(WIN32) || defined(_WIN32)

